### DISCUSSION TITLE
BT and linux

### DISCUSSION THREAD

**Discussion by vrossum** [cedricp/ddt4all#971] [Q&A]
Hi
Does anyone have luck in connecting via BT under linux?
Windows shows Comm4  and Comm5 ports, but nothing is shown under linux.

I can pair the device, but it doesn't connect, and gives the error
br-connection-profile-unavailable

**Comment by shrlnm** [comment]
@vrossum 
>ls -lrt /dev

**Comment by KarelSvo** [comment]
and

groups

**Comment by Furtif** [comment]
https://askubuntu.com/questions/58119/changing-permissions-on-serial-port

**Comment by davidwkerr** [comment]
I have successfully run ecutweaker version 1.85 on my KW902 BT OBD on Renault Zoe.

I have successfully built DDT4ALL on several Ubuntu systems. However, when I pair the KW902 in the Linux settings, I cannot see anything in DDT4ALL in the top window whether or not I click the BT icon in DDT4ALL.

When I click "Connected Mode" I get the message to select a serial port. But BT devices in Linux do not have a serial port! Also, the comments above are also referring to a serial port. I do not understand! What am I missing here? 

Any guidance would be much appreciated.

Many thanks,
David

**Reply by davidwkerr** [reply]
I think I shall need to create a /dev/rfcomm0 binding it to the dongle....

**Comment by davidwkerr** [comment]
Fixed!
As well as ECU Tweaker working, I now have DDT4ALL working with my dongles on a Ubuntu laptop.

Apart from building DDT4ALL, you need to take the following steps:

a. Pair your OBD device/s in the Settings->Bluetooth screen.

Take note of the dongle MAC addresses (let's call them MAC1 and MAC2 for two OBD dongles)

b. At the command line, create and map virtual devices /dev/rfcomm0 and /dev/rfcomm1 to the two dongles, as follows:

sudo rfcomm bind /dev/rfcomm0 MAC1
sudo rfcomm bind /dev/rfcomm1 MAC2

If you have trouble writing to the dongle, chmod 777 /dev/rfcomm* but creating the virtual entries should have given read and write to owner and group so should not be necessary- depending upon how your system is set up, sudoers/groups et al.

So, if you want to use Linux, the above is the last step needed to enable communication to the OBD dongle/s. You can select between dongles on the DDT4ALL startup scrfeen.
